pkg/ethereum/call_msg.go

```diff
@@ -4,18 +4,23 @@ import (
 	"encoding/json"
 	"math/big"
 
+	"github.com/ethereum/go-ethereum/core/types"
+
 	ethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 )
 
 // CallMsg contains parameters for contract calls.
 type CallMsg struct {
-	From     *ethcommon.Address // the sender of the 'transaction'
-	To       *ethcommon.Address // the destination contract (nil for contract creation)
-	Gas      *uint64            // if 0, the call executes with near-infinite gas
-	GasPrice *big.Int           // wei <-> gas exchange ratio
-	Value    *big.Int           // amount of wei sent along with the call
-	Data     *[]byte            // input data, usually an ABI-encoded contract method invocation
+	From       *ethcommon.Address // the sender of the 'transaction'
+	To         *ethcommon.Address // the destination contract (nil for contract creation)
+	Gas        *uint64            // if 0, the call executes with near-infinite gas
+	GasPrice   *big.Int           // wei <-> gas exchange ratio
```

Comment:
do you know why are we using big.Int here and hexutil.Big in the rest of Msg structs?

Reply:
The architecture of the JSON-RPC is not correct as everything is mixed between infra, service and business layer inside `pkg/ethereum`. Some calls are done to the underlying node (infra). Some are requests (JSON received in HTTP) and some are internal types (entities). We should fix this mess at some point. Here I only followed what was already implemented.

Reply:
If that is the case, would you mind to add a TODO with your findings here? Once you do that you can resolve this